eos/utils/expression.hh

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2021 Méril Reboud
+ *
+ * This file is part of the EOS project. EOS is free software;
+ * you can redistribute it and/or modify it under the terms of the GNU General
+ * Public License version 2, as published by the Free Software Foundation.
+ *
+ * EOS is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59 Temple
+ * Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+#ifndef EXPRESSION_HH
+#define EXPRESSION_HH 1
+
+#include <eos/observable-fwd.hh>
+#include <eos/utils/expression-fwd.hh>
+
+#include <cassert>
+#include <memory>
+#include <iostream>
+#include <map>
+
+namespace eos::exp
+{
+    class BinaryExpression
+    {
+        public:
+            char _op;
+            Expression _lhs, _rhs;
+
+            typedef double(*func)(double, double);
+
+            static double Add(double a, double b);
+            static double Subtract(double a, double b);
+            static double Multiply(double a, double b);
+            static double Divide(double a, double b);
+
+            static BinaryExpression::func Method(char op);
+
+            BinaryExpression() {}
+
+            BinaryExpression(char op, Expression const& l, Expression const& r) :

```

Comment:
EOS coding style puts ``const`` before the reference's type, i.e.: ``const Expression & l``, with all the spaces.